src/main/host/syscall/socket.c

```diff
@@ -377,8 +372,15 @@ static int _syscallhandler_setSocketOptHelper(SysCallHandler* sys, Socket* sock,
         case SO_RCVBUF: {
             const unsigned int* val =
                 process_getReadablePtr(sys->process, sys->thread, optvalPtr, sizeof(int));
-            size_t newsize =
-                (*val) * 2; // Linux kernel doubles this value upon setting
+            size_t newsize = (*val) * 2; // Linux kernel doubles this value upon setting
+
+            // Linux also has limits SOCK_MIN_RCVBUF (slightly greater than 2048) and the sysctl max
+            // limit. We choose some reasonable limits for Shadow. The minimum limit in man 7 socket
+            // is incorrect.
+            newsize = MAX(newsize, 2048);
+            // This upper limit was chosen as an arbitrarily high number.
```

Comment:
Again, a little bit of explanation about context and how we got to this number would be helpful in the future.